flow/controllers/imitation_learning/imitating_network.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import tensorflow as tf
+from utils import *
+import tensorflow_probability as tfp
+from flow.controllers.base_controller import BaseController
+from replay_buffer import ReplayBuffer
+
+
+class ImitatingNetwork():
+    """
+    Neural network which learns to imitate another given expert controller.
+    """
+    # Implementation in Tensorflow
+
+    def __init__(self, sess, action_dim, obs_dim, num_layers, size, learning_rate, replay_buffer_size, training = True, inject_noise=0, noise_variance=0.5, policy_scope='policy_vars'):
+
+        self.sess = sess
+        self.action_dim = action_dim
+        self.obs_dim = obs_dim
+        self.num_layers = num_layers
+        self.size = size
+        self.learning_rate = learning_rate
+        self.training = training
+        self.inject_noise=inject_noise
+        self.noise_variance = noise_variance
+
+        with tf.variable_scope(policy_scope, reuse=tf.AUTO_REUSE):
+            self.build_network()
```

Comment:
Why do you need an AUTO_REUSE here?

Reply:
I put an AUTO_REUSE here so that the same variables will be reused when the graph is rerun (so copies of the variables (weights/biases) don't get recreated)